tools/gyp/pylib/gyp/generator/make.py

```diff
@@ -31,6 +31,17 @@
 from gyp.common import GetEnvironFallback
 from gyp.common import GypError
 
+# hashlib is supplied as of Python 2.5 as the replacement interface for sha
+# and other secure hashes.  In 2.6, sha is deprecated.  Import hashlib if
+# available, avoiding a deprecation warning under 2.6.  Import sha otherwise,
+# preserving 2.4 compatibility.
+try:
+  import hashlib
```

Comment:
since GYP is kinda abandoned and you're floating the patch, you can remove all that and just import hashlib. Python 2.4 predates YouTube, keeping compatibility with that is insane these days.